### PR TITLE
fix mach spawn fail interrupt

### PR DESCRIPTION
**Pull request by vtjnash**
I believe this should fix the occasional failure we've seen this past year on the spawn test on macOS where it gets a spurious InterruptException generated by the mach kernel. I guess that's what the man page meant when it said this function could have "unspecified behavior".

### CODE REVIEW

File: deps/libuv.version
Comment (tkelman) [inline deps/libuv.version]: why a partial sha? 52d72a52cc7ccd570929990f010ed16e2ec604c8

File: src/signals-unix.c
Comment (yuyichao) [inline src/signals-unix.c]: This one shouldn't be necessary? Or is this just to help future debugging?

**Review by yuyichao — APPROVED** [review]

File: src/signals-unix.c
Comment (vtjnash) [inline src/signals-unix.c]: future debugging

File: deps/checksums/libuv-52d72a52cc7ccd570929990f010ed16e2ec604c.tar.gz/md5
Comment (tkelman) [inline deps/checksums/libuv-52d72a52cc7ccd570929990f010ed16e2ec604c.tar.gz/md5]: filenames are missing the last char

File: .travis.yml
Comment (tkelman) [inline .travis.yml]: Please do not merge this unless you've verified that this branch builds from a completely clean cache. The deps should not need the spawn modification.

**Review by tkelman — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
